src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestStream.cs

```diff
@@ -122,11 +144,11 @@ public override int Read(byte[] buffer, int offset, int count)
             throw new NotSupportedException();
         }
 
-        internal void EndUpload()
+        internal async Task EndUploadAsync(CancellationToken token)
```

Comment:
The `CopyToAsync` method is public and non-virtual and is on the `HttpContent` class. We can't "overload" that method for use with WinHttpHandler. So, we need this extra method to write the last zero-length chunk.